### PR TITLE
Improve student activity registration system

### PR DESCRIPTION
**Pull request by AndreGDFeliciano**
This pull request introduces several major improvements to the Mergington High School Activities app, focusing on new features for participant management, enhanced frontend interactivity, and comprehensive automated testing. The most important changes are grouped below.

**Backend enhancements:**

- Added a new API endpoint to allow students to unregister from activities, including proper validation and error handling for edge cases such as non-existent activities or students not registered for an activity.
- Improved the signup logic to prevent duplicate registrations by checking if a student is already signed up before adding them as a participant.

**Frontend improvements:**

- Updated the activities list UI to display current participants for each activity, including a styled section with participant emails and an "unregister" (delete) button for each participant. [[1]](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3L13-R56) [[2]](diffhunk://#diff-090bb81494e72cfcac6b0a6065f88e9caf9d80c00b887e38f02a690dbec0f8b2R77-R129)
- Implemented interactive unregister functionality: users can now remove participants directly from the UI, with confirmation dialogs, error/success messaging, and automatic refreshing of the activities list. [[1]](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3R67-R121) [[2]](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3R143-R145)

**Testing and reliability:**

- Added a comprehensive suite of automated tests for all API endpoints, including edge cases and end-to-end workflows for signing up and unregistering participants.
- Introduced pytest fixtures to ensure the in-memory activities database is reset before and after each test, maintaining test isolation and reliability.
- Added module docstrings to test files for clarity and maintainability.

**Data and initialization:**

- Expanded the initial activities database with additional sports, artistic, and intellectual activities, each with sample participants, to provide a richer demo and testing environment.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    
+    # Reset to original state before each test
+    activities.clear()
+    activities.update(original_activities)
+    
+    yield
+    
+    # Reset after test as well
+    activities.clear()
+    activities.update(original_activities)
```
[the remark this answers is not in the document]
Comment (AndreGDFeliciano) [reply tests/conftest.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/AndreGDFeliciano/flai-workshop-github-copilot-100/pull/2#discussion_r2787310588)

[the remark this answers is not in the document]
Comment (AndreGDFeliciano) [reply tests/conftest.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/AndreGDFeliciano/flai-workshop-github-copilot-100/pull/2#discussion_r2787310588)
